HTTPServer.core/Server.cs

```diff
@@ -80,30 +81,49 @@ private List<byte> PullDataFromClient()
             var messageReceived = new List<byte>();
             while (true)
             {
-                var bytesReceived = ReadKbOfData(messageReceived);
-                if (IsLessThanKb(bytesReceived))
+                ReadKbOfData(messageReceived);
+                if (IsDoneReading(messageReceived))
                     break;
             }
             return messageReceived;
         }
 
-    private static bool IsLessThanKb(int bytesReceived)
-    {
-        return bytesReceived < 1024;
-    }
+        private bool IsDoneReading(List<byte> messageReceived)
+        {
+            var messageBytes = FormatObtainedData(messageReceived);
+            var message = Encoding.UTF8.GetString(messageBytes);
+            return message.Contains("\r\n\r\n") && HasWholeBody(message);
+        }
 
-    private int ReadKbOfData(List<byte> messageReceived)
+        private bool HasWholeBody(string message)
         {
-            var buffer = new byte[1024];
-            var bytesReceived = ReceiveData(buffer); 
-            for (var bufferIndex = 0; bufferIndex < bytesReceived; bufferIndex++)
-                messageReceived.Add(buffer[bufferIndex]);
-            return bytesReceived;
+            if(!message.Contains("Content-Length:"))
+                return true;
+            var headers = message.Split(new string[] { "\r\n" }, StringSplitOptions.None);
+            var contentLength = CalculateContentLength(headers);
+            var headerBodySplit = message.Split(new string[] { "\r\n\r\n" }, StringSplitOptions.None);
+            return contentLength <= headerBodySplit[1].Length;
         }
 
-        private int ReceiveData(byte[] buffer)
+        private int CalculateContentLength(string[] headers)
         {
-            return _clientConnection.Receive(buffer);
+            for (int i = 0; i < headers.Length - 1; i++)
+            {
+                if (headers[i].Contains("Content-Length:"))
+                {
+                    var startOfNumber = headers[i].IndexOf("Content-Length:") + 16;
```

Comment:
I'm a little confused which part you are saying is brittle.  It does handle white space before and after "Content-Length:".  For example, the header line could be "Content-Length:       40" and it will still work.  Parse("       40") will give an int value of 40, because it handles whitespace.